src/lighteval/models/endpoints/inference_providers_model.py

```diff
@@ -0,0 +1,244 @@
+    proxies: Any | None = None
+    generation_parameters: GenerationParameters = field(default_factory=GenerationParameters)
+
+    @classmethod
+    def from_path(cls, path):
+        with open(path, "r") as f:
+            config = yaml.safe_load(f)["model"]
+
+        model = config["model_name"]
+        provider = config.get("provider", None)
+        timeout = config.get("timeout", None)
+        proxies = config.get("proxies", None)
+        generation_parameters = GenerationParameters.from_dict(config)
+        return cls(
+            model=model,
+            provider=provider,
+            timeout=timeout,
+            proxies=proxies,
+            generation_parameters=generation_parameters,
+        )
+
+
+class InferenceProvidersClient(LightevalModel):
+    def __init__(self, config) -> None:
+        """
+        IMPORTANT: Your API keys should be set in the environment variables.
+        If a base_url is not set, it will default to the public API.
+        """
+
+        self.model_info = ModelInfo(
+            model_name=config.model,
+            model_sha="",
+            model_dtype=None,
+            model_size="",
+        )
+        self.model = config.model
+        self.provider = config.provider
+        self.generation_parameters = config.generation_parameters
+
+        self.API_MAX_RETRY = 5
+        self.API_RETRY_SLEEP = 3
+        self.API_RETRY_MULTIPLIER = 2
+        self.pairwise_tokenization = False
+
+        self.client = AsyncInferenceClient(
+            provider=self.provider,
+            timeout=config.timeout,
+            proxies=config.proxies,
+        )
+        self._tokenizer = AutoTokenizer.from_pretrained(self.model)
+
+    def _encode(self, text: str):
+        enc = self._tokenizer(text=text)
+        return enc
+
+    def tok_encode(self, text: str | list[str]):
+        if isinstance(text, list):
+            toks = [self._encode(t["content"]) for t in text]
+            toks = [tok for tok in toks if tok]
+            return toks
+        return self._encode(text)
+
+    async def __call_api(self, prompt, num_samples):
+        """Make API call with retries."""
+        for attempt in range(self.API_MAX_RETRY):
+            try:
+                kwargs = {
+                    "model": self.model,
+                    "messages": prompt,
+                    "n": num_samples,
+                }
+                kwargs.update(self.generation_parameters.to_inference_providers_dict())
+                response: ChatCompletionOutput = await self.client.chat.completions.create(**kwargs)
+                return response
+            except Exception as e:
+                wait_time = min(64, self.API_RETRY_SLEEP * (2**attempt))  # Exponential backoff with max 64s
+                logger.warning(
+                    f"Error in API call: {e}, waiting {wait_time} seconds before retry {attempt + 1}/{self.API_MAX_RETRY}"
+                )
+                time.sleep(wait_time)
```

Comment:
using `time.sleep()` in an async function blocks the entire event loop, preventing all other coroutines from running so let's avoid using it here.
```suggestion
                await asyncio.sleep(wait_time)
```

Reply:
Oh thanks for the catch !